packages/core/src/publicCredential/PublicCredential.spec.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2018-2022, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * @group unit/publicCredential
+ */
+
+import type {
+  AssetDidUri,
+  DidUri,
+  IAssetClaim,
+  IClaimContents,
+  IPublicCredential,
+} from '@kiltprotocol/types'
+
+import { BN } from '@polkadot/util'
+import { ApiMocks } from '@kiltprotocol/testing'
+import { ConfigService } from '@kiltprotocol/config'
+import { devAlice } from '../__integrationtests__/utils'
+import * as CType from '../ctype'
+import * as PublicCredential from '../publicCredential'
+
+const testCType = CType.fromProperties('raw ctype', {
+  name: { type: 'string' },
+})
+
+const mockApi = ApiMocks.createAugmentedApi()
+ConfigService.set({ api: mockApi })
+
+// Build a public credential with fake attestation (i.e., attester, block number, revocation status) information.
+function buildCredential(
+  assetDid: AssetDidUri,
+  contents: IClaimContents
+): IPublicCredential {
+  const claim: IAssetClaim = {
+    cTypeHash: CType.idToChain(testCType.$id),
+    contents,
+    subject: assetDid,
+  }
+  const credential = PublicCredential.fromClaim(claim)
+  const attester: DidUri = `did:kilt:${devAlice.address}`
```

Comment:
Is there a way to generate a DidUri which is not so coupled to the implementation details?